lib/networks/known_networks.go

```diff
@@ -1103,15 +1103,15 @@ func SetNetworks(networkCfg config.NetworkConfig) ([]blockchain.EVMNetwork, erro
 		}
 		// if network is not simulated or forked, use the rpc urls and wallet keys from config
 		if !strings.Contains(networkName, "SIMULATED") && !forked {
-			var ok bool
-			wsUrls, ok = networkCfg.RpcWsUrls[selectedNetworks[i]]
-			if !ok {
-				return nil, fmt.Errorf("no rpc ws urls found in config for '%s' network", selectedNetworks[i])
-			}
-
-			httpUrls, ok = networkCfg.RpcHttpUrls[selectedNetworks[i]]
-			if !ok {
-				return nil, fmt.Errorf("no rpc http urls found in config for '%s' network", selectedNetworks[i])
+			var ok, wsOk, httpOk bool
+			// Check for WS URLs
+			wsUrls, wsOk = networkCfg.RpcWsUrls[selectedNetworks[i]]
+			// Check for HTTP URLs
+			httpUrls, httpOk = networkCfg.RpcHttpUrls[selectedNetworks[i]]
+
+			// Ensure at least one of WS or HTTP URLs is available
+			if !wsOk && !httpOk {
+				return nil, fmt.Errorf("no RPC URLs found for '%s' network; at least one HTTP or WS RPC URL must be set", selectedNetworks[i])
 			}
```

Comment:
this will fail if no ws endpoints are provided, no? why not use the same condition as in `lib/config/network.go`?